### DISCUSSION TITLE
Bootc to be contributed to the CNCF as a Sandbox project!

### DISCUSSION THREAD

**Discussion by cgwalters** [bootc-dev/bootc#897] [Announcements]
We're excited to announce that bootc [is submitted](https://github.com/cncf/sandbox/issues/310) to the [CNCF](https://cncf.io) as a sandbox project!

A core goal for this project is to normalize using containers and cloud native technology all the way down to the booted host system, and CNCF includes many projects which are aligned with this at a higher level, from container-native CI/CD frameworks to KubeVirt.

With this contribution, we plan to officially join that community and participate directly \- benefiting from shared support, resources and best practices in the CNCF. Red Hat has provided very significant resources in the development of this project and that will continue \- this project is included as part of [RHEL Image Mode](https://www.redhat.com/en/technologies/linux-platforms/enterprise-linux/image-mode). But at the same time a goal with this contribution is to provide an explicitly vendor-neutral home to extend the reach of bootc and collaborate with the other CNCF projects. We’d love to build out support for more operating systems, integrations with Kubernetes (such as update orchestrators, cluster API, etc.) as well as generic Linux environments, and in general build cloud native infrastructure with you!  

More links:

- https://github.com/cncf/sandbox/issues/310
- https://www.redhat.com/en/blog/red-hat-contribute-comprehensive-container-tools-collection-cloud-native-computing-foundation
- https://blog.podman.io/2024/11/podman-buildah-and-skopeo-and-the-cloud-native-computing-foundation/

**Comment by cgwalters** [comment]
We were accepted into sandbox! https://github.com/cncf/sandbox/issues/310#issuecomment-2595023366

**Comment by cgwalters** [comment]
Followup tasks in https://github.com/cncf/sandbox/issues/339

Sandbox reference issue: [#310](https://github.com/cncf/sandbox/issues/310)

## REQUIRED BEFORE PROCEEDING WITH ONBOARDING

A "Project Contribution Agreement" must be completed and any existing trademarks **MUST** be transferred to the Linux Foundation **BEFORE** further project onboarding tasks can be completed.

- [ ] Review and understand the [CNCF IP Policy](https://github.com/cncf/foundation/blob/main/charter.md#11-ip-policy). Ensure you are using a CNCF compatible license; inbound projects must use the Apache 2.0 license or seek approval for exceptions. Licenses for dependencies are covered separately below.
- [x] Review and understand the [LF trademark guidelines](https://www.linuxfoundation.org/legal/trademark-usage).
- [ ] Transfer any [trademark and logo assets to the Linux Foundation](https://github.com/cncf/foundation/tree/main/agreements) via the Contribution Agreement. CNCF staff will send this document to the contact emails listed in the Sandbox application.

---

## @cgwalters Review and understand other documents

- [x] The [Technical Leadership Principles](https://github.com/cncf/toc/blob/main/PRINCIPLES.md#technical-leadership-principles) that outlines the expected behavior for any maintainer in a leadership role.
- [x] The [project proposal process and requirements](https://github.com/cncf/toc/blob/main/process/README.md).
- [x] The [services available for your project at the CNCF](https://contribute.cncf.io/resources/project-services/).
- [x] The [dependency license allowlist](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist).
- [x] The [online program guidelines](https://github.com/cncf/foundation/blob/main/online-programs-guidelines.md).
- [x] The [telemetry data collection and usage policy](https://www.linuxfoundation.org/legal/telemetry-data-policy).
- [x] [Book time with CNCF staff](http://project-meetings.cncf.io) to understand project benefits and event resources.


## @jeckersb  Review and understand other documents

- [x] The [Technical Leadership Principles](https://github.com/cncf/toc/blob/main/PRINCIPLES.md#technical-leadership-principles) that outlines the expected behavior for any maintainer in a leadership role.
- [x] The [project proposal process and requirements](https://github.com/cncf/toc/blob/main/process/README.md).
- [x] The [services available for your project at the CNCF](https://contribute.cncf.io/resources/project-services/).
- [x] The [dependency license allowlist](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist).
- [x] The [online program guidelines](https://github.com/cncf/foundation/blob/main/online-programs-guidelines.md).
- [x] The [telemetry data collection and usage policy](https://www.linuxfoundation.org/legal/telemetry-data-policy).
- [x] [Book time with CNCF staff](http://project-meetings.cncf.io) to understand project benefits and event resources.

## @ckyrouac  Review and understand other documents

- [x] The [Technical Leadership Principles](https://github.com/cncf/toc/blob/main/PRINCIPLES.md#technical-leadership-principles) that outlines the expected behavior for any maintainer in a leadership role.
- [x] The [project proposal process and requirements](https://github.com/cncf/toc/blob/main/process/README.md).
- [x] The [services available for your project at the CNCF](https://contribute.cncf.io/resources/project-services/).
- [x] The [dependency license allowlist](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist).
- [x] The [online program guidelines](https://github.com/cncf/foundation/blob/main/online-programs-guidelines.md).
- [x] The [telemetry data collection and usage policy](https://www.linuxfoundation.org/legal/telemetry-data-policy).
- [x] [Book time with CNCF staff](http://project-meetings.cncf.io) to understand project benefits and event resources.

## Contribute and transfer other materials

- [x] Move your project to its own separate neutral GitHub organization. This will make it transferable to the CNCF's GitHub Enterprise account. If it's already in a GHE account, you will need to remove it from that first.
- [x] Accept the invite to join the CNCF GitHub Enterprise account. We'll then add `thelinuxfoundation` as an organization owner to ensure neutral hosting of your project.
- [x] Migrate your Slack channels (if any) to the [Kubernetes or CNCF Slack workspace](https://slack.com/help/articles/217872578-Import-data-from-one-Slack-workspace-to-another). CNCF staff can help.
- [x] Join the [#maintainers-circle](https://app.slack.com/client/T08PSQ7BQ/C014YQ8CDCG) Slack channel.
- [ ] Transfer [your domain to the CNCF](https://jira.linuxfoundation.org/plugins/servlet/desk/portal/2/create/63?requestGroup=19). The "LF Stakeholder email" is <projects@cncf.io>. The "Project" is CNCF.
- [ ] Submit a [pull request](https://github.com/cncf/artwork) with your artwork. (xref https://gitlab.com/fedora/design/community-design-team/issues/-/issues/158#note_2470834299)
- [ ] Transfer website analytics to <projects@cncf.io>. CNCF staff can help.

## Update and document project details

- [x] Ensure that [DCO](https://github.com/apps/dco) (preferred) or [CLA](https://github.com/cncf/cla) are enabled for all GitHub repositories of the project.
- [x] Ensure that that the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md) (or your adopted version of it) are explicitly referenced at the project's `README.md` on GitHub. (@jeckersb )
- [x] Ensure LF footer is on your website and [guidelines](https://github.com/cncf/foundation/blob/main/website-guidelines.md) are followed (if your project doesn't have a dedicated website, please adopt those guidelines for the `README.md` file). The project-specific footer text you should use will be provided in the Contribution Agreement instructions. (@jmarrero )
- [x] Create a maintainer list and add it to the [aggregated CNCF maintainer list](https://maintainers.cncf.io) via pull request: https://github.com/cncf/foundation/pull/987
- [x] Provide emails for the maintainers to get access to the maintainers mailing list and Service Desk. Email them to <project-onboarding@cncf.io>. (@gursewak1997 )
- [x] Start working on [written, open governance](https://contribute.cncf.io/maintainers/governance/) and consider adding this to a `GOVERNANCE.md` file at the root of your repo. TODO (@jmarrero REF https://github.com/containers/podman/pull/25398 )
- [x] Start on an [OpenSSF Best Practices Badge](https://www.bestpractices.dev/en). (@gursewak1997 )

## CNCF staff tasks to support the project

- [x] Add the project to [DevStats](https://devstats.cncf.io/).
- [ ] Add the project to [CLOmonitor](https://clomonitor.io/).
- [ ] Add the project to [LFX Insights](https://insights.lfx.linuxfoundation.org/). This is done by adding a read-only app to your GitHub organization once it's in CNCF GHE.
- [ ] Add the project to [LFX Project Control Center](https://projectadmin.lfx.linuxfoundation.org/project/a0941000002wBz4AAE).
- [ ] Add the project to the [Cloud Native Landscape](https://landscape.cncf.io) by updating [landscape.yml](https://github.com/cncf/landscape/blob/master/landscape.yml) following these [instructions](https://github.com/cncf/landscape2/blob/main/docs/config/data.yml).
- [ ] Add a license scanning tool, like [FOSSA](https://fossa.com/) or [Snyk](https://snyk.io/).
- [ ] Send a welcome email to confirm maintainer list access.

**Reply by cgwalters** [reply]
cc @jeckersb @ckyrouac since you are listed as maintainers along with me we all need to go through this.

**Reply by cgwalters** [reply]
I split the "review the docs" into a section for each of us

**Comment by cgwalters** [comment]
# Thread on github organization


One of the biggest questions here is the Github organization. It's unfortunate that https://github.com/bootc is taken. (Heh and now that I look, [so is podman](https://github.com/podman)). Anyways...this may partially depend on what the podman folks choose to do over in [their issue](https://github.com/cncf/sandbox/issues/338#issuecomment-2605454656).

Options:

- bootc-dev
- bootc-org

**Reply by cgwalters** [reply]
I've reserved bootc-dev. Last call for naming.

**Reply by jmarrero** [reply]
The only argument I can see for bootc-org is if we own https://bootc.org/

If not bootc-dev is OK

**Reply by jeckersb** [reply]
bootc.org isn't us, whois shows it was registered in 2010 and has nameservers for ns1.bootc.uk ns2.bootc.net, which then leads to https://www.bootc.net/ (but Chris Boot *is* apparently a Debian developer!)

**Reply by henrywang** [reply]
`bootc.dev` is available. `bootc-dev` makes sense.

**Comment by cgwalters** [comment]
# Thread on project services

Let's discuss in this sub-thread which [project services](https://contribute.cncf.io/resources/project-services/) we use. I think it would make sense to have a bootc meeting on the CNCF hosted zoom.

- zoom:
- https://fossa.io/ looks like a no-brainer to use
- slack: oh yeah...hard decision here on whether to create a new channel on the CNCF slack or not...
- https://www.gitpod.io/ - maybe? Could be interesting to try out

**Comment by cgwalters** [comment]
Update: I am going to try to make some progress on this tomorrow, which would probably include the Github repository move etc.

**Comment by cgwalters** [comment]
Transfer is complete! https://github.com/bootc-dev/bootc

**Reply by cgwalters** [reply]
Followup https://github.com/bootc-dev/bootc/issues/1180

**Comment by cgwalters** [comment]
# Thread on domain name 

So @p4 got bootc.io - ref https://github.com/cncf/sandbox/issues/339#issuecomment-2605472101

That said there's [a hazy future for .io](https://thenewstack.io/what-is-the-future-of-the-io-domain/) and it probably makes sense to use something generic like `bootc.dev`? I'm fine with that. But we could keep bootc.io and just make it a redirect to bootc.dev?

**Reply by jmarrero** [reply]
I also like the idea of having bootc.dev for the mental map of bootc-dev to bootc.dev

**Reply by p5** [reply]
It's @p5 with the domain :) 

The .io domains are used by a lot of the other large CNCF projects, so thought it was a good fit.  I understand the concerns about the future, though there's going to be much larger problems than bootc's website not being available if it is taken away.

Some examples: 
- podman.io
- kubernetes.io
- prometheus.io
- opentelemetry.io
- cncf.io
- fluxcd.io

The domain is not used for anything except redirecting to the docs currently.

**Reply by castrojo** [reply]
bootc.dev was available so I snagged it.

**Reply by cgwalters** [reply]
@p5 Hard problem in computer science: off by one errors :smile:  Thanks so much again for grabbing that domain!! It definitely makes sense to use and I agree.

> bootc.dev was available so I snagged it.

Awesome!! :+1: I think let's submit both to the CNCF list for transfer...but we need to bikeshed which is canonical. My vote is probably to make bootc.dev canonical because of the rationale that it matches the github org and because of the lingering .io issue.

**Reply by cgwalters** [reply]
@castrojo can you start the process of transferring the domain? I think that and the artwork are the last two things

**Reply by castrojo** [reply]
Yeah someone ping me on slack I forgot how to do this. 😄

**Reply by cgwalters** [reply]
The step info is
> Transfer [your domain to the CNCF](https://jira.linuxfoundation.org/plugins/servlet/desk/portal/2/create/63?requestGroup=19). The "LF Stakeholder email" is [projects@cncf.io](mailto:projects@cncf.io). The "Project" is CNCF.

Does that help?

**Reply by castrojo** [reply]
I should have known this. 😄 Filed.

The LF IT ticket number is: IT-28121

**Reply by castrojo** [reply]
Ok domain is transferred into LFX, I'll follow up on how you can self-manage that next time we sync.

**Comment by cgwalters** [comment]
This is now done! https://github.com/cncf/sandbox/issues/339#issuecomment-2932207696 
There was a long road due to some boring mechanical legal issues among other things, but done now!
